watertap/unit_models/electrodialysis_1D.py

```diff
@@ -1070,7 +1072,7 @@ def calculate_scaling_factors(self):
                     )
                     ** 0.5
                 )
-                sf_eleosm = Constants.faraday_constant
+                sf_eleosm = Constants.faraday_constant * pyunits.mol / pyunits.coulomb
```

Comment:
As far as I can tell, this was not caught because we checked units before calculating scaling factors. The scaling factor is applied to the constraint after it is calculated, so maybe we need to check units after scaling from now on.